at-quannguyen/W3/dictionary.rb

```diff
@@ -0,0 +1,111 @@
+require 'time'
+# class dictionary
+class Dictionary
+  def initialize
+    @list_dictionary = {}
+    @id = 1
+  end
+
+  def add(dict, des)
+    if Dictionary.check_empty_by_vocab(@list_dictionary, dict)
+      @list_dictionary[@id] =
+        { dict: dict, des: des, lookup: 0, history_lookup: 0 }
+      @id += 1
+      return true
+    end
+    false
+  end
+
+  def lookup(vocab)
+    result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+    if result
+      sleep 0.5
+      result[1][:lookup] += 1
+      result[1][:history_lookup] = Time.new.strftime('%m/%d/%Y %H:%M:%S')
+      return result[1][:des]
+    end
+    nil
+  end
+
+  def size
+    @list_dictionary.length
+  end
+
+  def remove(vocab)
+    result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+    check = result ? result[0] : nil
+    @list_dictionary.delete(check) ? true : false
+  end
+
+  def pop
+    @list_dictionary.delete(@list_dictionary.keys.max)
+  end
+
+  def update(vocab, desc)
+    unless Dictionary.check_empty_by_vocab(@list_dictionary, vocab)
+      result = Dictionary.find_dict_by_vocab(@list_dictionary, vocab)
+      @list_dictionary.delete(result[0])
+      return add(vocab, desc)
+    end
+    false
+  end
+
+  def random
+    keys = @list_dictionary.keys
+    key_rand = keys[rand(keys.size)]
+    result = @list_dictionary[key_rand]
+    { result[:dict] => result[:des] }
+  end
+
+  def favorite
+    arr = []
+    result = nil
+    lookup = @list_dictionary.values.map { |e| e[:lookup] }
+    ck_favorite(arr, lookup, result)
+  end
+
+  def ck_favorite(arr, lookup, result)
+    unless lookup.max.zero?
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/6f22e2e17b7a320ef0dcf3e430a8fa8270c1e786/at-quannguyen/W3/dictionary.rb#L68"></span>Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals)</td>
    </tr>
  </tbody>
</table>